content/exchange/artifacts/Windows.Audit.CISCat_Lite.yaml

```diff
@@ -0,0 +1,62 @@
+name: Windows.Audit.CISCat_Lite
+description: |
+  This artifact runs the CISCat-Lite tool on the target machine and uploads the html output on the velociraptor server.
+  
+  CIS-CAT Lite is a free tool from the Center for Internet Security that helps in assessing and improving IT security configurations. 
+  It enables unlimited system scans, provides compliance scores, and offers remediation steps based on CIS Benchmarks. 
+  This tool is useful for organizations seeking to enhance their technology security.
+
+  Downloads your CIS-CAT Lite free from the https://learn.cisecurity.org/cis-cat-lite link. (Registration is required!)
+  After downloading the zip file, upload the file to the velociraptor server as the CISCat_Lite tool. 
+  The artifact has been configured to perform a standard scan of a Windows 10 Enterprise machine. 
+  To select the baseline and profiles, execute the command ".\Assessor-CLI.bat -i"
+type: CLIENT
+
+author: Antonio Blescia (TheThMando)
+
+parameters:
+  - name: BaselinePath
+    default: "./benchmarks/CIS_Microsoft_Windows_10_Enterprise_Benchmark_v2.0.0-xccdf.xml"
+  - name: ProfileName
+    default: "Level 1 (L1) - Corporate/Enterprise Environment (general use)"
+
+tools:
+  - name: CISCat_Lite
+    serve_locally: true
+
+sources:
+  - precondition: SELECT OS From info() where OS = 'windows'
+
+    query: |
+      -- Generate the temp dir
+      LET TmpDir <= tempdir(remove_last=true)
+      -- Upload the CISCat_Lite tool on the host
+      LET Toolzip <= SELECT FullPath
+        FROM Artifact.Generic.Utils.FetchBinary(ToolName="CISCat_Lite",
+                                                IsExecutable=FALSE)
+      -- Extract the CISCat_Lite tool zip
+      LET _ <= SELECT *
+        FROM unzip(filename=Toolzip.FullPath, output_directory=TmpDir)
+      -- Generate the absolute path that points the the extracted tool location  
+      LET CISCatPath = path_join(
+          components=[TmpDir, 'Assessor', 'Assessor-CLI.bat'],
+          path_type='windows')
+      -- Compose the command line string     
+      LET cmdline <= join(
```

Comment:
you should not build command lines like this - use an array directly to execve() 

This is vulnerable to shell injection.

Reply:
Dear @scudette, I used the same logic implemented inside the PersistenceSniper Artifact: https://github.com/Velocidex/velociraptor-docs/blob/86271e14350473049618b5d6f9b7875cdd11210f/content/exchange/artifacts/Windows.Forensics.PersistenceSniper.yaml#L76

Reply:
I know that the **BaselinePath** and **ProfileName** can be used to inject shell commands; but how I can permit to a user to define this value from the artifact parameters?
Can you suggest to me an artifact as reference?
Thanks

Reply:
Do you prefer this form?
```
-- Run the command using PowerShell     
LET _ <= SELECT *
FROM execve(argv=["powershell", "-command", './Assessor-CLI.bat', '-b', '"' + BaselinePath + '"', '-p', '"' + ProfileName + '"'], cwd=TmpDir + '/' + "Assessor")
```

Reply:
you dont need the powershell -command at all. you also dont need quotes when using the argv approach

Reply:
The persistance hunter artifact is also fragile for the same reason (if the directory contains a quote it will break) but it uses powershell to pipe to csv so i guess it really needs to call to powershell.

In this case it does not appear to need powershell at all so it is better to do it properly.

Reply:
you dont need to quote anything when passing directly to execve - quoting is only needed when passing to the shell.

Reply:
Thank you @scudette, I learn another lesson about execve!
I replace the code with the below:
```
-- Run the command using powershell     
LET _ <= SELECT *
FROM execve(argv=[TmpDir + '/' + "Assessor"+'/'+"Assessor-CLI.bat", "-b", BaselinePath, '-p', ProfileName])
```
If it is ok, I push the code to my branch!
Thanks a lot!

Reply:
Maybe `TmpDir + '/Assessor/Assessor-CLI.bat"`

Reply:
I solved using the **CISCatPath** variable. 👍

Reply:
I solved using the CISCatPath variable. 👍